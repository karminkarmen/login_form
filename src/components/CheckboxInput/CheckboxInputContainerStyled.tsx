import styled from 'styled-components';
import { centerized } from '../../configs/mixins';

const CheckboxInputContainerStyled = styled.div`
    ${centerized}
    margin: ${props => props.theme.margins.normal};
`;

export default CheckboxInputContainerStyled;