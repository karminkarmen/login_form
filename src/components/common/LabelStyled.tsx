import styled from 'styled-components';

type Props = {
    target: string;
    isUppercased: boolean;
}

const LabelStyled = styled.label.attrs((props: Props) => ({
    htmlFor: props.target
}))<Props>`
    margin: ${props => props.theme.margins.normal};
    color: ${props => props.theme.colors.main};
    text-transform: ${props => props.isUppercased ? 'uppercase' : 'none'};
    font-weight: bold;
    letter-spacing: ${props => props.theme.letterSpacing.normal};
    font-size: ${props => props.theme.fontSizes.normal};
    font-family: ${props => props.theme.fontFamilies.main}, sans-serif;
`;

export default LabelStyled;
