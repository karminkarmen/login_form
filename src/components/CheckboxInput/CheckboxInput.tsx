import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../../configs/theme';
import LabelStyled from '../common/LabelStyled';
import CheckboxInputContainerStyled from './CheckboxInputContainerStyled';

export type Props = {
    name: string;
    label: string;
}

const CheckboxInput: React.FC<Props> = ({ name, label }) => (
    <ThemeProvider theme={theme}>
        <CheckboxInputContainerStyled>
            <input type="checkbox" name={name} id={name} />
            <LabelStyled 
                target={name} 
                isUppercased={false} 
            >
                {label}
            </LabelStyled>
        </CheckboxInputContainerStyled> 
    </ThemeProvider>
);

export default CheckboxInput;
