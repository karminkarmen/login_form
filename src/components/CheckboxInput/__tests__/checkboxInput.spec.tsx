import React from 'react';
import { shallow } from 'enzyme';

import CheckboxInput from '../CheckboxInput';
import LabelStyled from '../../common/LabelStyled';
import CheckboxInputContainerStyled from '../CheckboxInputContainerStyled';

describe('<CheckboxInput />', () => {

    const props: React.ComponentProps<typeof CheckboxInput> = {
        name: 'remember',
        label: 'Remember me',
    };

    it('matches snapshot', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should render input with checkbox type', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        expect(wrapper.find(`input[type='checkbox']`)).toHaveLength(1);
    });

    it('should render input with specified name and id', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        expect(wrapper.find(`input[name='${props.name}']`)).toHaveLength(1);
        expect(wrapper.find(`input[id='${props.name}']`)).toHaveLength(1);
    });

    it('should render styled label', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        expect(wrapper.find(LabelStyled)).toHaveLength(1);
    });

    it('should render styled label connected with checkbox', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        const label = wrapper.find(LabelStyled);
        expect(label.find({ target: props.name })).toHaveLength(1);
    });

    it('should render styled label which contains provided text', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        const label = wrapper.find(LabelStyled);
        expect(label.text()).toEqual(props.label);
    });

    it('should render styled contianer', () => {
        const wrapper = shallow(<CheckboxInput {...props} />);

        expect(wrapper.find(CheckboxInputContainerStyled)).toHaveLength(1);
    });
});
