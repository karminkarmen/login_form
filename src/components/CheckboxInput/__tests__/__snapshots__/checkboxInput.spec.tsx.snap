// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxInput /> matches snapshot 1`] = `
"<ThemeProvider theme={{...}}>
  <styled.div>
    <input type=\\"checkbox\\" name=\\"remember\\" id=\\"remember\\" />
    <styled.label target=\\"remember\\" isUppercased={false}>
      Remember me
    </styled.label>
  </styled.div>
</ThemeProvider>"
`;
